### PR TITLE
Migrating to csproj

### PR DESCRIPTION
**Pull request by sebastienros**
To start tracking the changes and talk.

### CODE REVIEW

**Comment by Jetski5822** [comment]
I can comment wherever :P

So the issue we have at the moment is that VS2017 doesnt support Orchard , right? do we have an ETA on that?

File: src/Orchard.Admin.Abstractions/Orchard.Admin.Abstractions.csproj [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (Jetski5822) [inline src/Orchard.Admin.Abstractions/Orchard.Admin.Abstractions.csproj]: Where are the default namespace settings? i.e. this one should be Orchard.Admin

File: src/Orchard.BackgroundTasks.Abstractions/Orchard.BackgroundTasks.Abstractions.csproj [hunk 2]
```diff
@@ -0,0 +1,23 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <VersionPrefix>2.0.0</VersionPrefix>
+    <TargetFramework>netstandard1.6</TargetFramework>
+    <DebugType>portable</DebugType>
+    <AssemblyName>Orchard.BackgroundTasks.Abstractions</AssemblyName>
+    <PackageId>Orchard.BackgroundTasks.Abstractions</PackageId>
+    <PackageTargetFallback>$(PackageTargetFallback);dnxcore50</PackageTargetFallback>
```
Comment (Jetski5822) [inline src/Orchard.BackgroundTasks.Abstractions/Orchard.BackgroundTasks.Abstractions.csproj]: Lets get rid of this... $(PackageTargetFallback);

File: src/Orchard.Cms.Web/Modules/Lucene/Lucene.csproj [hunk 3]
```diff
@@ -0,0 +1,31 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <VersionPrefix>2.0.0</VersionPrefix>
+    <TargetFramework>netstandard1.6</TargetFramework>
+    <WarningsAsErrors>true</WarningsAsErrors>
+    <DebugType>portable</DebugType>
+    <AssemblyName>Lucene</AssemblyName>
+    <PackageId>Lucene</PackageId>
+    <PackageTargetFallback>$(PackageTargetFallback);dnxcore50;portable-net451+win8</PackageTargetFallback>
```
Comment (Jetski5822) [inline src/Orchard.Cms.Web/Modules/Lucene/Lucene.csproj]: Do we need ;portable-net451+win8?

File: src/Orchard.Cms.Web/Modules/Lucene/Lucene.csproj [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    <GenerateAssemblyConfigurationAttribute>false</GenerateAssemblyConfigurationAttribute>
```
Comment (Jetski5822) [inline src/Orchard.Cms.Web/Modules/Lucene/Lucene.csproj]: Can we remove GenerateAssemblyConfigurationAttribute? what does it do?

File: src/Orchard.Indexing.Abstractions/Orchard.Indexing.Abstractions.csproj [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    <GenerateAssemblyProductAttribute>false</GenerateAssemblyProductAttribute>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\Orchard.ContentManagement.Abstractions\Orchard.ContentManagement.Abstractions.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    
+  </ItemGroup>
```
Comment (Jetski5822) [inline src/Orchard.Indexing.Abstractions/Orchard.Indexing.Abstractions.csproj]: empty item group

**Comment by Jetski5822** [comment]
Remove all these...

GenerateAssemblyCompanyAttribute
GenerateAssemblyProductAttribute
GenerateAssemblyConfigurationAttribute

**Comment by sebastienros** [comment]
https://blogs.msdn.microsoft.com/visualstudio/2017/01/26/update-to-visual-studio-2017-release-candidate/

**Comment by agriffard** [comment]
I installed VS 2017 RC3.
When I open the solution, the projects are all empty.

Do the files have to be included in each project to compile?

**Comment by sebastienros** [comment]
@agriffard that's super weird! It works just fine for me, and the automatic migration didn't have to do it. Can you paste the exact version number you are using (from Help/About)

**Comment by agriffard** [comment]
Microsoft Visual Studio Enterprise 2017 RC
Version 15.0.26020.0 D15REL

**Comment by sebastienros** [comment]
See this:
http://www.natemcmaster.com/blog/2017/01/19/project-json-to-csproj/

> NB: many of default globbing patterns are added automatically by Microsoft.NET.Sdk and Microsoft.NET.Sdk.Web.

It should be implicit. Maybe you are having a bug that needs to be reported. I will ask. Send me a screenshot.

**Comment by agriffard** [comment]
He doesn't say it is implicit.

**Comment by agriffard** [comment]
Without `<Compile Include="**\*.cs" />`:
image


With `<Compile Include="**\*.cs" />`:
image

**Comment by DanielStolt** [comment]
I have the same version number as you @agriffard and I have not updated. Shouldn't your build number be higher?

**Comment by Jetski5822** [comment]
Do we really need this?

```
  <ItemGroup>
    <Compile Include="**\*.cs" />
  </ItemGroup>
```

**Comment by sebastienros** [comment]
@agriffard you have an old version. Or you have multiple version and are not using the latest.

**Comment by agriffard** [comment]
OK, my bad. I updated again and I now have Version 15.0.26127.0 D15REL.

I can see the projects.
Only TheBlogTheme was unable to be loaded but I managed to re-migrate it. 

When I launch in debug, I have an error:

NullReferenceException: Object reference not set to an instance of an object.
in Orchard.Environment.Extensions.ExtensionManager.cs line 353
`                    foreach (var type in otherExtension.ExportedTypes)`
when it tries to load 'Orchard.Tokens.Content'

**Comment by sebastienros** [comment]
Forgot to mention that this doesn't work, just sharing for people to work on it with what has been done so far.

**Comment by jtkech** [comment]
And extensions compilation / loading need to be re-worked because right now it relies on the presence of project.json files. But i don't have VS2017 installed.

For TheBlogTheme it works because it doen't need any assembly and its exported types. So, just an internal `extension.IsError` is set to true.

**Pull request closed without merging**